changelog/unreleased/issue-2202

```diff
@@ -0,0 +1,16 @@
+Enhancement: Add upload checksum for azure, gs, s3 and swift backends
+
+Up to now only the B2 backend and partially the Swift backend verified the
+integrity of uploaded files. The verification works by informing the backend
+about the expected hash of the uploaded data. This allows the backend to verify
```

Comment:
Maybe stress that the hash is calculated on the *encrypted* data?

Reply:
Isn't that clear from the context (all data stored in the backend is encrypted)?

Reply:
I've slightly tweaked the wording to mention that the uploaded files are encrypted. And `uploaded data` is now replaced with `uploaded files`.